baseline-refaster-testing/src/main/java/com/palantir/baseline/refaster/RefasterTestHelper.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.refaster;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.CodeTransformer;
+import com.google.errorprone.apply.DescriptionBasedDiff;
+import com.google.errorprone.apply.ImportOrganizer;
+import com.google.errorprone.apply.SourceFile;
+import com.google.errorprone.refaster.RefasterRuleBuilderScanner;
+import com.google.testing.compile.JavaFileObjects;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.util.TreePath;
+import com.sun.tools.javac.tree.JCTree;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import javax.tools.JavaFileObject;
+import org.assertj.core.api.Assertions;
+import org.junit.rules.ExternalResource;
+
+public final class RefasterTestHelper extends ExternalResource {
+
+    private final List<CodeTransformer> transformers;
+
+    public static RefasterTestHelper forRefactoring(Class<?> refasterRuleClass) {
+        return new RefasterTestHelper(refasterRuleClass);
+    }
+
+    private RefasterTestHelper(Class<?> refasterRuleClass) {
+        Path sourceFile = Paths
+                .get("src/main/java")
+                .resolve(refasterRuleClass.getName().replaceAll("\\.", File.separator) + ".java");
```

Comment:
should probably make this hack clear in the docstring so people don't expect magic

```
The source code of the given refaster rule should exist in {@code src/main/java} under the right package path
```

Reply:
Can you put it on the static constructor at least? :P
This method is not discoverable.